src/renderer/src/components/Sidebar/SettingsSidebarLink.tsx

```diff
@@ -0,0 +1,32 @@
+import { cloneElement, useEffect, useState } from 'react'
+
+export const SettingsSidebarLink = ({ id, icon, title, onClick, selectedId }): JSX.Element => {
+  const [isActive, setIsActive] = useState<boolean>(id === selectedId)
```

Comment:
It is better to receive a prop called `active` and who is rendering it should pass this prop, for example: 

```
<SettingsSidebarLink
    title={t('personalisationOption.networkConfiguration')}
    icon={<Tb3DCubeSphere />}
    onClick={() => handlePersonalisationClick(SettingsPersonalisationOption.NETWORK_CONFIGURATION)}
    active={selectedPersonalisationOption === SettingsPersonalisationOption.NETWORK_CONFIGURATION}
```

And you do that, you wont need the useState, useEffect